controllers/clusterversion_controller.go

```diff
@@ -320,6 +348,22 @@ func (c *ClusterVersionReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	return ctrl.Result{}, nil
 }
 
+func (c *ClusterVersionReconciler) deletionPhase() (reconcile.Result, error) {
+	if err := csi.DeleteCSIDriver(c.ctx, c.Client, csi.GetCephFSDriverName()); err != nil && !k8serrors.IsNotFound(err) {
+		c.log.Error(err, "unable to delete cephfs CSIDriver")
+		return ctrl.Result{}, err
+	}
+	if err := csi.DeleteCSIDriver(c.ctx, c.Client, csi.GetRBDDriverName()); err != nil && !k8serrors.IsNotFound(err) {
+		c.log.Error(err, "unable to delete rbd CSIDriver")
+		return ctrl.Result{}, err
+	}
+	if err := c.Client.Delete(c.ctx, c.scc); err != nil && !k8serrors.IsNotFound(err) {
+		c.log.Error(err, "unable to delete SCC")
+		return ctrl.Result{}, err
+	}
```

Comment:
I am not sure if Security Context Constraints resources are cluster-scopped resources or namespace-scoped resources. If they are namespace scope we can have an owner reference on them which will remove the need for this code

Reply:
from what i've checked it is a cluster scoped resource so that is not a possibility unfortunately.

Reply:
Ok , resolving